code-challenges.test.js

```diff
@@ -17,35 +17,81 @@
 const people = [
   { name: "ford prefect", occupation: "a hitchhiker" },
   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
+  { name: "arthur dent", occupation: "a radio employee" },
+];
 // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 
+// a describe method that lists the name of the function OR naming of the particular test.
+describe("capName", () => {
+  // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+  it("capName will capitalize the name and join the occupation and return the sentence in an array.", () => {
+    //an expect method, nested within the test block, calling on the hello() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+    expect(
+      capName(string.charAt(0).toUpperCase() + string.slice(1).join)
+    ).toEqual("Ford Prefect is a hitchhiker.");
+    expect(
+      capName(string.charAt(0).toUpperCase() + string.slice(1).join)
+    ).toEqual("Zaphod Beeblebrox is president of the galaxy.");
+    expect(
+      capName(string.charAt(0).toUpperCase() + string.slice(1).join)
+    ).toEqual("Arthur Dent is a radio employee.");
+  });
+});
 
 // b) Create the function that makes the test pass.
 
+function capitalizeFirstLetter(string) {
+  returncstring.charAt(0).toUpperCase() + string.slice(1).join;
+}
+
+console.log(capitalizeFirstLetter("foo")); // Foo
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false];
 // Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true];
 // Expected output: [ 2, 1, -1 ]
 
+// a describe method that lists the name of the function OR naming of the particular test.
+describe("mixedData", () => {
+  // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+  it("returns an array without the words", () => {
+    //an expect method, nested within the test block, calling on the hello() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+    expect(mixedData("[^0-9]", "", a_string)).toEqual([2, 0, -1, 0]);
+    expect(mixedData("[^0-9]", "", a_string)).toEqual([2, 1, -1]);
```

Comment:
`expect(mixedData(hodgepodge1)).toEqual([2, 0, -1, 0])`
An expect statement calls the name of the function and passes in the test variable. Then you expect it to equal the given output.